GitHubTeams.ps1

```diff
@@ -452,62 +804,648 @@ function New-GitHubTeam
+
+    .EXAMPLE
+        Remove-GitHubTeam -OrganizationName PowerShell -TeamName Developers
+
+        Removes the 'Developers' GitHub team from the 'PowerShell' organization.
+
+    .EXAMPLE
+        Remove-GitHubTeam -OrganizationName PowerShell -TeamName Developers -Force
+
+        Removes the 'Developers' GitHub team from the 'PowerShell' organization without prompting.
+
+    .EXAMPLE
+        $team = Get-GitHubTeam -OrganizationName PowerShell -TeamName Developers
+        $team | Remove-GitHubTeam -Force
+
+        You can also pipe in a GitHub team that was returned from a previous command.
+#>
+    [CmdletBinding(
+        SupportsShouldProcess,
+        PositionalBinding = $false,
+        ConfirmImpact = 'High',
+        DefaultParameterSetName = 'TeamSlug')]
+    [Alias('Delete-GitHubTeam')]
+    param
+    (
+        [Parameter(
+            Mandatory,
+            ValueFromPipeline,
+            ValueFromPipelineByPropertyName,
+            Position = 1)]
+        [ValidateNotNullOrEmpty()]
+        [string] $OrganizationName,
+
+        [Parameter(
+            Mandatory,
+            ValueFromPipeline,
+            ValueFromPipelineByPropertyName,
+            Position = 2,
+            ParameterSetName='TeamName')]
+        [ValidateNotNullOrEmpty()]
+        [string] $TeamName,
+
+        [Parameter(
+            Mandatory,
+            ValueFromPipelineByPropertyName,
+            ParameterSetName='TeamSlug')]
+        [ValidateNotNullOrEmpty()]
+        [string] $TeamSlug,
+
+        [switch] $Force,
+
+        [string] $AccessToken
+    )
+
+    Write-InvocationLog
+
+    $telemetryProperties = @{
+        OrganizationName = (Get-PiiSafeString -PlainText $RepositoryName)
+        TeamSlug = (Get-PiiSafeString -PlainText $TeamSlug)
+        TeamName = (Get-PiiSafeString -PlainText $TeamName)
+    }
+
+    if ($PSBoundParameters.ContainsKey('TeamName'))
+    {
+        $getGitHubTeamParms = @{
+            OrganizationName = $OrganizationName
+            TeamName = $TeamName
+        }
+        if ($PSBoundParameters.ContainsKey('AccessToken'))
+        {
+            $getGitHubTeamParms['AccessToken'] = $AccessToken
+        }
+
+        $team = Get-GitHubTeam @getGitHubTeamParms
+        $TeamSlug = $team.slug
+    }
+
+    $uriFragment = "/orgs/$OrganizationName/teams/$TeamSlug"
+
+    if ($Force -and (-not $Confirm))
+    {
+        $ConfirmPreference = 'None'
+    }
+
+    if (-not $PSCmdlet.ShouldProcess($TeamName, 'Remove Github Team'))
+    {
+        return
+    }
+
+    $params = @{
+        UriFragment = $uriFragment
+        Method = 'Delete'
+        Description =  "Deleting $TeamSlug"
+        AccessToken = $AccessToken
+        TelemetryEventName = $MyInvocation.MyCommand.Name
+        TelemetryProperties = $telemetryProperties
+    }
+
+    Invoke-GHRestMethod @params | Out-Null
+}
+
+filter Remove-GitHubTeamProject
+{
+<#
+    .SYNOPSIS
+        Removes a team project from an organization on GitHub.
```

Comment:
This description is actually incorrect, which was confusing for me at first.  I had to read the API to better understand what's happening here, and now that I have, I think the API needs to be renamed as well (although I'm still not sure on the correct name yet).

The correct description for this is:

```suggestion
Removes an organization project from a team on GitHub. This command removes the project from the team, but does not delete the project.
```

This API will not cause the project to be removed.  It's only changing the _permission_ such that the indicated team no longer has access to it.  That's also where this API name becomes confusing.  At first I was going to tell you that it shouldn't be its own API and should instead just be part of `Remove-GitHubProject` with additional parameters...but this _isn't_ removing the project....it's just altering the permissions on it.